x-pack/solutions/observability/plugins/synthetics/server/routes/pings/journey_screenshot_blocks.ts

```diff
@@ -19,16 +19,31 @@ export const createJourneyScreenshotBlocksRoute: SyntheticsRestApiRouteFactory =
     }),
   },
   writeAccess: false,
-  handler: async ({ request, syntheticsEsClient }) => {
-    const { hashes: blockIds } = request.body;
+  handler: async ({ request, response, syntheticsEsClient }) => {
+    const { hashes: blockIds } = request.body as { hashes: string[] };
 
     const result = await getJourneyScreenshotBlocks({
       blockIds,
       syntheticsEsClient,
     });
 
-    return {
-      result,
-    };
+    if (result.length === 0) {
+      return response.noContent({
+        body: result,
+      });
+    }
+    const resultSet = new Set(result.map((block) => block.id));
+    if (blockIds.some((id: string) => !resultSet.has(id))) {
+      return response.multiStatus({
+        body: blockIds.map((id) =>
+          resultSet.has(id)
+            ? { ...result.find((block) => block.id === id), status: 200 }
+            : { id, status: 404 }
+        ),
+      });
```

Comment:
How will UI process these multi-status results?

Reply:
Good point - I made a minor addition to the client api file. See https://github.com/elastic/kibana/pull/210937/commits/82f6bd69502e718698654c593785da5c7a971796.

Behavior on `main` for missing block:
![20250214095327](https://github.com/user-attachments/assets/4e59cee9-e1d1-4c56-b259-8798b1167aa0)

Behavior on this branch:
![20250214105600](https://github.com/user-attachments/assets/3396e23f-f2d5-4ad9-a7fa-4ecfab459abd)

The behavior is the same as if we received an empty set. We should do another follow-up to show a better placeholder when we get a `noContent` or `multiStatus` response.